### PR TITLE
Implement functionality for adding/removing a card to favorites

### PR DESCRIPTION
**Pull request by dreyhiflden**
functionality for adding/removing a card to favorites and show total favorite cards

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+          <button @click="$emit('remove-this', card)">Remove this card</button> <!-- removeThis(card) -->
+          <button v-if="!card.isHidden && card.subtitle" @click="$emit('hide-subtitle', card)">Hide subtitle</button>
+          <button v-if="card.isHidden" @click="$emit('show-subtitle', card)">Show subtitle</button>
+          <button v-if="!card.isHidden && !isFavorite(card)" @click="$emit('add-to-favorites', card)">Add to fav</button>
+          <button v-if="!card.isHidden && isFavorite(card)" @click="$emit('remove-from-favorites', card)">Remove from fav</button>
+        </div>
+      `,
+  methods: {
+    isFavorite(card) {
+      return this.$root.isFavorite(card);
```
Comment (carrot-e) [inline index.js]: Although it's possible to access the parent this way, it's considered a bad practice. Think how you can solve this without using `$root`.

File: index.js [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
+    },
+  },
+};
+
 new Vue({
   el: '#app',
+  components: {
+    'component-card': componentCard
```
Comment (carrot-e) [inline index.js]: This isn't critical, but FYI: usual naming convention is `<name>Component`.

**Comment by carrot-e** [comment]
Also, a bug:
when you remove a "favorite" card, the favorites counter isn't updated:
Screenshot from 2019-04-15 17-32-11

File: index.js (hunk 2 above)
Comment (dreyhiflden) [reply index.js]: Rename

File: index.js (hunk 1 above)
Comment (dreyhiflden) [reply index.js]: Rework to use props instead $root

**Pull request merged**